include/boost/url/bnf/parse.hpp

```diff
@@ -77,214 +96,176 @@ parse(
     error_code& ec,
     char ch) noexcept;
 
-/** @brief Parse a sequence of elements
+/** Parse a substring into one or more BNF elements
 
    This function parses the string defined by the first chars in the
-   range of chars [@ref it, @ref end) into the Backus–Naur form (BNF)
-   elements [arg_0, arg_1, ..., arg_n] in @ref args.
-
-   For each type Arg_i in [Arg_0, Arg_1, ..., Arg_n] in @ref Args, a
-   corresponding function overload
-   `bool parse(char const*& it, char const* const end, error_code&, Arg_i const& t)`
-   should be defined to consume the input from [@ref it, @ref end) into `arg_i`.
-
-   This `parse` function is defined for `char`s so that it matches the BNF
-   string literals in [@ref it, @ref end).
-
-   The function returns `true` and resets @ref ec when the input is
-   parsed successfully. If the input cannot be parsed into the BNF elements,
-   the function returns `false` and sets the error number in @ref ec to the
-   appropriate error number defined in @ref boost::urls::error.
+   range of chars [`it`, `end`) into the Backus–Naur form (BNF)
+   elements in `args`. For each `arg` in `args`, a  corresponding function overload of `parse`
+   `bool parse(char const*& it, char const* const end, error_code&, Arg const& arg)`
+   should be defined to consume the input from [`it`, `end`).
```

Comment:
This needs an example for a hypothetical user-defined type `T`.

Reply:
This is what we had before, but my understanding was that you thought it was too long. Any recommendations?

Reply:
Yes we can sit together and work on it.